### DISCUSSION TITLE
How to get edit extension to work?

### DISCUSSION THREAD

**Discussion by flschr** [datenstrom/community#625] [Ask a question] [answered]
I installed the edit extension for a much smoother update process. Unfortunately I can't login to my site. I created (and approved) a user account, but every time when I try to login to my website, the user/password fields got cleared and nothing is happening at all. Whats wrong here? Can you help me out?

**Comment by flschr** [comment]
What I already tried:
- Different browsers (also private profile)
- Acccess from mobile phone (with cellular connection) to make sure that the error isn't caused by my local wifi
- New / different user
- Reset of "extension" folder from clean setup

... but the error still occurs. Can anybody help me out?

**Reply by flschr** [reply]
Does anybody have a suggestion what to check? I currently don't know how to savely upgrade my existing installation :(

**Reply by schulle4u** [reply]
Can you provide more information about your setup? E. g. operating system, web server, php version. Also check your `system/extensions/yellow.log` for possible installation errors and post them here. 

See also the [troubleshooting guide](https://datenstrom.se/yellow/help/troubleshooting)

**Reply by flschr** [reply]
Of course, I will try to provide the requested informations. I use [this webspace package](https://www.manitu.de/webhosting/webhosting-m/) from the German hoster Manitu.

Tech specs I was able to find out:
- OS of the webserver: unknown Linux
- Webserver: Apache
- PHP version: 8.0 (but also tried 7.3, 7.4)
- The yellow.log doesn't contain any errors (see my last entries)

**yellow.log**
> 2021-01-16 07:04:16 error Can't parse file 'system/extensions/comments.php'!
2021-01-16 15:53:24 info Install extension 'Comments 0.8.16'
2021-03-22 10:03:35 info Install extension 'Edit 0.8.42'


I read through the troubleshooting guide, so far everything looks ok for me.

**Reply by schulle4u** [reply]
> `2021-01-16 07:04:16 error Can't parse file 'system/extensions/comments.php'!`

Is this extension still installed?

**Reply by flschr** [reply]
Nope, I deleted it a few weeks ago (by removing all comments.* files and deleting all comments related entries inside the yellow-system.ini). There is als no current comments related error in the log file anymore.

**Reply by flschr** [reply]
Any chance to debug the login process? I've tried debug-1/3 but didn't recognize any errors.

**Comment by IngoBelka** [comment]
What about Apache error log? What info do you get in yellow.log by increasing debug level? Look at [troubleshooting guide](https://datenstrom.se/yellow/help/troubleshooting), how to do that, just mentioned by schulle4u.

**Reply by flschr** [reply]
Actually no errors in the Apache log. I really don't know whats causing the problems. Most likely its a somehow akward configuration of my hosting package.

Anyway, as I only need the edit extension for updating Yellow I ask you guys if the following steps would work as a manual update?

1. When you guys announce a update, download the latest release from https://github.com/datenstrom/yellow/archive/master.zip
2. Rename (and later delete) my extensions folder and copy the new one from "master.zip"
3. Add "yellow-user.ini" from renamed extension folder and add my modified values from my backup.
4. Add my plugins (in my case only 3 php files) from renamed extension folder and copy the extension entries from my old "yellow-system.ini" to the new one.
5. As I use Yellow as a blog, I also transfer "blog.html" and "blogpages.html" to the layouts folder (and add my customizations again).

**Comment by IngoBelka** [comment]
Could you install a developement environment on your own local system? There you can make all things you like, and after you finished, you can upload to your web host. That might be the better way. For me this works instantly and I have a good workflow.

**Reply by flschr** [reply]
Yeap, this is how I will proceed in the future. Thanks for your help.

**Comment by flschr** [comment, marked as answer]
As the edit extension still isn't work for me, I was looking for a easy solution for maintaining my Yellow based blog (esp. creating content and updating Yellow). In the meantime I stumbled upon the idea of using a Github repository as powerhouse for Yellow. I created [a public repository](https://github.com/flschr/Recycled-Words) with a extensive readme.md that describes everything in detail. That solves the issue of the not working edit extension so for me and also has a lot of other benefits as well. If you have questions, feel free to ask.
